paasta_tools/cli/cmds/logs.py

```diff
@@ -422,65 +490,94 @@ def tail_logs(self, service, levels, components, clusters, raw_mode=False):
+            kw = {
+                'scribe_env': scribe_env,
+                'stream_name': get_log_name_for_service(service),
+                'service': service,
+                'levels': levels,
+                'components': components,
+                'clusters': clusters,
+                'queue': queue,
+                'filter_fn': paasta_log_line_passes_filter,
+            }
+            process = Process(target=self.scribe_tail, kwargs=kw)
+            spawned_processes.append(process)
+            process.start()
+
+        def marathon_fn(scribe_env, cluster):
+            kw = {
+                'scribe_env': scribe_env,
+                'stream_name': 'stream_marathon_%s' % cluster,
+                'service': service,
+                'levels': levels,
+                'components': components,
+                'clusters': [cluster],
+                'queue': queue,
+                'parse_fn': parse_marathon_log_line,
+                'filter_fn': marathon_log_line_passes_filter,
+            }
+            process = Process(target=self.scribe_tail, kwargs=kw)
+            spawned_processes.append(process)
+            process.start()
+
+        def chronos_fn(scribe_env, cluster):
+            kw = {
+                'scribe_env': scribe_env,
+                'stream_name': 'stream_chronos_%s' % cluster,
+                'service': service,
+                'levels': levels,
+                'components': components,
+                'clusters': [cluster],
+                'queue': queue,
+                'parse_fn': parse_chronos_log_line,
+                'filter_fn': chronos_log_line_passes_filter,
+            }
+            process = Process(target=self.scribe_tail, kwargs=kw)
+            spawned_processes.append(process)
+            process.start()
+
+        def stdout_fn(scribe_env):
+            kw = {
+                'scribe_env': scribe_env,
+                'stream_name': get_log_name_for_service(service, prefix='app_output'),
+                'service': service,
+                'levels': levels,
+                'components': components,
+                'clusters': clusters,
+                'queue': queue,
+                'filter_fn': paasta_log_line_passes_filter,
+            }
+            process = Process(target=self.scribe_tail, kwargs=kw)
+            spawned_processes.append(process)
+            process.start()
+
+        def stderr_fn(scribe_env):
+            kw = {
+                'scribe_env': scribe_env,
+                'stream_name': get_log_name_for_service(service, prefix='app_output'),
+                'service': service,
+                'levels': levels,
+                'components': components,
+                'clusters': clusters,
+                'queue': queue,
+                'filter_fn': paasta_log_line_passes_filter,
+            }
+            process = Process(target=self.scribe_tail, kwargs=kw)
+            spawned_processes.append(process)
+            process.start()
```

Comment:
All of these seem the same except for `kw` and whether they take a `cluster` param. How about this refactor:

 - Make a single function that takes parameters `component, scribe_env, cluster` and looks up `kw` based on component
 - make run_code_over_scribe_envs call that function for each `component, scribe_env, cluster`